xdsl/dialects/builtin.py

```diff
@@ -1725,132 +1726,61 @@ def get_element_type(self) -> _UnrankedMemrefTypeElems:
 @irdl_attr_definition
 class DenseIntOrFPElementsAttr(TypedAttribute, ContainerType[AnyDenseElement]):
     name = "dense"
-    type: ParameterDef[
-        RankedStructure[IntegerType]
-        | RankedStructure[IndexType]
-        | RankedStructure[AnyFloat]
-    ]
-    data: ParameterDef[ArrayAttr[AnyIntegerAttr] | ArrayAttr[AnyFloatAttr]]
```

Comment:
Wouldn't it make sense to migrate ArrayAttr to be bytes-based instead of DenseIntOrFPElementsAttr, BTW?

Reply:
I'm not 100% sure of this BTW, but would be good to play with it and maybe chat about findings. It would be nice to have some sort of generic way to have a dense array of things, but maybe that should just be this class, and array should be left alone to explicitly be a tuple of attributes and nothing more.